tests/bytes.rs

```diff
@@ -691,48 +674,52 @@ fn create_input_coin_data_offset() {
-                            static_contracts.clone(),
-                            storage_slot.clone(),
-                            inputs,
-                            outputs.clone(),
-                            witnesses.clone(),
-                        );
-
-                        let mut tx_p = tx.clone();
-                        tx_p.precompute_metadata();
-
-                        buffer.iter_mut().for_each(|b| *b = 0x00);
-                        let _ = tx
-                            .read(buffer.as_mut_slice())
-                            .expect("Failed to serialize input");
-
-                        let (offset, len) = tx
-                            .input_coin_predicate_offset(last_input)
-                            .expect("Failed to fetch offset");
-
-                        let (offset_p, _) = tx_p
-                            .input_coin_predicate_offset(last_input)
-                            .expect("Failed to fetch offset from tx with precomputed metadata!");
-
-                        assert_eq!(offset, offset_p);
-                        assert_eq!(predicate.as_slice(), &buffer[offset..offset + len]);
-                    }
+    for storage_slot in storage_slots.iter() {
+        for inputs in inputs.iter() {
+            for outputs in outputs.iter() {
+                for witnesses in witnesses.iter() {
+                    let mut inputs = inputs.clone();
+                    let last_input = inputs.len();
+                    inputs.push(input_coin.clone());
+
+                    dbg!(&inputs);
+
+                    let mut tx = Transaction::create(
+                        gas_price,
+                        gas_limit,
+                        byte_price,
+                        maturity,
+                        bytecode_witness_index,
+                        salt,
+                        storage_slot.clone(),
+                        inputs,
+                        outputs.clone(),
+                        witnesses.clone(),
+                    );
+
+                    let mut tx_p = tx.clone();
+                    tx_p.precompute_metadata();
+
+                    dbg!(&tx_p.metadata());
+
+                    buffer.iter_mut().for_each(|b| *b = 0x00);
+                    let _ = tx
+                        .read(buffer.as_mut_slice())
+                        .expect("Failed to serialize input");
+
+                    let (offset, len) = tx
+                        .input_coin_predicate_offset(last_input)
+                        .expect("Failed to fetch offset");
+
+                    let (offset_p, _) = tx_p
+                        .input_coin_predicate_offset(last_input)
+                        .expect("Failed to fetch offset from tx with precomputed metadata!");
+
+                    assert_eq!(offset, offset_p);
+                    assert_eq!(
+                        predicate.as_slice(),
+                        &buffer[offset..offset + len][..predicate.len()]
```

Comment:
Unsure if this is correct

Reply:
hmm it seems like we shouldn't need `[..predicate.len()]` since `offset..offset + len` should already be enough to slice the predicate bytes.

Reply:
For some reason `tx.input_coin_predicate_offset(last_input)` returns `8` as the length, but `predicate.len()` returns `2`. I've not yet managed to find what causes this disparency.

Reply:
Ok, so apparently the length of a coin predicate is padded to a word-size in length calculation, but predicate bytes themselves aren't. https://github.com/FuelLabs/fuel-tx/blob/8e9dde715701f4535c07c2538c5f8ba7a4754c0a/src/transaction/types/input.rs#L226

Reply:
Does `predicate.len()` return the length in instructions, or byte?

Reply:
Predicate is just a `Vec<u8>`, so the length is in bytes

Reply:
We have an open issue to discuss that. I think it should be fixed in a dedicated PR for that issue

https://github.com/FuelLabs/fuel-specs/issues/357